frontend/src/pages/projects/screens/spawner/dataConnection/useNotebookDataConnection.ts

```diff
@@ -15,36 +20,33 @@ export const getNotebookDataConnection = (
 };
 
 export const useNotebookDataConnection = (
+  dataConnections: DataConnection[],
   notebook?: NotebookKind,
-  dataConnections: DataConnection[] = [],
 ): [
   dataConnection: DataConnectionData,
-  setDataConnection: (connection: DataConnectionData) => void,
+  setDataConnection: UpdateObjectAtPropAndValue<DataConnectionData>,
+  resetDefaults: () => void,
 ] => {
-  const [dataConnection, setDataConnection] = React.useState<DataConnectionData>({
+  const createDataState = useGenericObjectState<DataConnectionData>({
     type: 'creating',
     enabled: false,
   });
+  const [, setCreateData] = createDataState;
 
-  React.useEffect(() => {
-    if (notebook) {
-      // find data connection from env list
-      const notebookDataConnectionSecret = getNotebookDataConnection(notebook, dataConnections)
-        ?.data.metadata.name;
+  const notebookDataConnectionSecret = getNotebookDataConnection(notebook, dataConnections)?.data
+    .metadata.name;
 
-      if (notebookDataConnectionSecret) {
-        setDataConnection({
-          type: 'existing',
-          enabled: true,
-          existing: {
-            secretRef: {
-              name: notebookDataConnectionSecret,
-            },
-          },
-        });
-      }
+  React.useEffect(() => {
+    if (notebookDataConnectionSecret) {
+      setCreateData('type', 'existing');
+      setCreateData('enabled', true);
+      setCreateData('existing', {
+        secretRef: {
+          name: notebookDataConnectionSecret,
+        },
+      });
     }
-  }, [notebook, dataConnections]);
+  }, [notebook, setCreateData, notebookDataConnectionSecret]);
```

Comment:
It's odd that once the default form data has been established, that we may still reset the values on the user in this `useEffect`. See previous comment as to whether or not we can remove this.